sktime/forecasting/residual_booster.py

```diff
@@ -0,0 +1,137 @@
+"""
+Implements a residual boosting forecaster.
+
+Which is an easy way to turn a forecaster without exogenous capability into one with.
+"""
+
+# copyright: sktime developers, BSD-3-Clause
+
+__all__ = ["ResidualBoostingForecaster"]
+__author__ = ["Sanchay117"]
+
+from sklearn.base import clone
+
+from sktime.forecasting.base import BaseForecaster
+
+
+class ResidualBoostingForecaster(BaseForecaster):
+    """
+    Residual boosting: add exogenous support to any base forecaster.
+
+    Parameters
+    ----------
+    base_forecaster : sktime forecaster
+        Point-forecast model that may ignore X.
+    residual_forecaster : sktime forecaster
+        Model trained on the base model's in-sample residuals.
+
+
+    Example
+    -------
+    >>> from sktime.datasets import load_airline
+    >>> from sktime.forecasting.naive import NaiveForecaster
+    >>> from sktime.forecasting.arima import ARIMA
+    >>> from sktime.forecasting import ResidualBoostingForecaster
+    >>> y = load_airline()
+    >>> fh = [1, 2, 3]
+    >>> booster = ResidualBoostingForecaster(
+    ...     base_forecaster=ARIMA(order=(1, 0, 0)),
```

Comment:
this is a "boring" forecaster, since the residual forecaster does not add much.

How about you showcase `YtoX` together with an endogenous forecaster that ignores `X`? Like the chronos case, but optimally do not use chronos since it is a heavy foundation model.